internal/ignition/discovery.go

```diff
@@ -209,6 +209,10 @@ func (ib *ignitionBuilder) shouldAppendOKDFiles(ctx context.Context, infraEnv *c
 	if cfg.OKDRPMsImage != "" {
 		return cfg.OKDRPMsImage, true
 	}
+	// Late-binding InfraEnvs will not have release images associated with them
+	if infraEnv.ClusterID == "" {
+		return "", false
+	}
 	// Check if selected payload contains `okd-rpms` image
 	releaseImage, err := ib.versionHandler.GetReleaseImage(ctx, infraEnv.OpenshiftVersion, infraEnv.CPUArchitecture, infraEnv.PullSecret)
```

Comment:
This should also probably be changed to use the cluster release versions, right?

Reply:
Good question, but wouldn't a bound infra-env's openshift version/arch match the cluster it's bound to or is that an incorrect assumption?

Reply:
I think you're right, but this just looked weird to me.

Reply:
To be clear, now that you mentioned the matching for bound clusters it's probably not worth changing it (especially because it looks like we'd need to plumb the db client into this function somehow).

Reply:
Yeah I thought getting the cluster from the db would be excessive, especially given that the version/arch should match. But I do agree that it does look funny.